psi4/driver/broker.py

```diff
@@ -0,0 +1,145 @@
+#
+# @BEGIN LICENSE
+#
+# Psi4: an open-source quantum chemistry software package
+#
+# Copyright (c) 2013-2018 The Psi4 Developers.
+#
+# The copyrights for code used from other parties are included in
+# the corresponding files.
+#
+# This file is part of Psi4.
+#
+# Psi4 is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, version 3.
+#
+# Psi4 is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License along
+# with Psi4; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# @END LICENSE
+#
+
+import string
+import sys
+import time
+import numpy as np
+from numpy.testing import assert_array_almost_equal as assert_equal
+
+import psi4
+try:
+    from ipi.interfaces.sockets import Client
+    ipi_available = True
+except ImportError:
+    ipi_available = False
+    # Define Client to enable testing of the Broker in the unittests
+    class Client(): pass
+
+
+class IPIBroker(Client):
+    def __init__(self, options, genbas=None, serverdata=False,
+                 molecule=None):
+        self.serverdata = serverdata
+        if not ipi_available:
+            psi4.core.print_out("i-pi is not available for import: ")
+            psi4.core.print_out("The broker infrastructure will not be available!\n")
+            super(IPIBroker, self).__init__()
+        elif serverdata:
+            mode, address, port = serverdata.split(":")
+            mode = string.lower(mode)
+            super(IPIBroker, self).__init__(address=address, port=port,
+                                      mode=mode)
+        else:
+            super(IPIBroker, self).__init__(_socket=False)
+        self.options = options
+
+        if molecule is None:
+            molecule = psi4.core.get_active_molecule()
+        self.initial_molecule = molecule
+        assert self.initial_molecule.orientation_fixed() == True, "Orientation must be fixed!"
```

Comment:
note that you can do things like https://github.com/psi4/psi4/blob/master/psi4/driver/procrouting/proc.py#L1442-L1446 to keep the c1 and orientation that the code needs without cooperation from the user.